### DISCUSSION TITLE
Coupling AeroDyn to an external structural solver written in Python

### DISCUSSION THREAD

**Discussion by wlllind** [OpenFAST/openfast#1926] [Model Integration]
I am looking for a way to couple AeroDyn to an external structural solver. Preferably, I want to use Python to call AeroDyn at a timestep, use the aerodynamic forces and moments in the structural solver, update the rotor-geometry and continue in that way until the last time step is reached. 

Of course, this is exactly how AeroDyn-ElastoDyn work. But, the structural solver is written in Python and allows me to add features that I can't add (due to lack of proficiency) in F90. Therefore

1. How does one achieve a way to call AeroDyn and 'pause' the simulation after every timestep using Python?  
2. Is there a step-by-step manual for achieving this?

I have already compiled the standalone AeroDyn driver. Moreover, aerodyn/python-lib/aerodyn_inflow_library.py might be of help, but I do not know how.

**Comment by jjonkman** [comment]
Dear @wlllind,

I agree that this is how AeroDyn is coupled to ElastoDyn, etc.  And I agree that if your structural solver is written in Python, that the existing Python wrapper around the AeroDyn and InflowWind modules of OpenFAST is what you should use.

The four main function calls in this Python wrapper (which are similar to the main Fortran SUBROUTINE calls of each module of OpenFAST) are:
aerodyn_inflow_init - Called at the beginning to initialize the module
aerodyn_inflow_updateStates - Called at each time step to updates the states of AeroDyn to the next time step
aerodyn_inflow_calcOutput - Called at each time step to calculates the outputs of AeroDyn at that step
aerodyn_inflow_end - Called at the end to terminate the module

Best regards,
